src/main/kotlin/com/chromaticnoise/multiplatformswiftpackage/task/CreateXCFrameworkTask.kt

```diff
@@ -1,32 +1,237 @@
+                val frameworkName = targets[0].name.value
+                val destinationDir = buildDir.resolve("bin/macosUniversal/${buildType}Framework")
+                mkdir(destinationDir.parent)
+                mkdir(destinationDir)
+                val tempUniversalBinaryLocation = File(destinationDir, frameworkName)
+                exec {
+                    commandLine(
+                        "lipo",
+                        "-create",
+                        "${targets[0].outputFile.path}/Versions/A/${frameworkName}",
+                        "${targets[1].outputFile.path}/Versions/A/${frameworkName}",
+                        "-output",
+                        tempUniversalBinaryLocation.path
+                    )
+                }
+                val aFramework = targets[0].outputFile.path
+                copy {
+                    from(aFramework)
+                    into(File(destinationDir, "$frameworkName.framework"))
+                }
+                // delete the old mono framework binary we copied here
+                val binaryFinalLocation = File(destinationDir, "${frameworkName}.framework/Versions/A/$frameworkName")
+                val binaryFinalSymlinkLocation = File(destinationDir, "${frameworkName}.framework/$frameworkName")
+                delete(binaryFinalLocation)
+                delete(binaryFinalSymlinkLocation)
+                // copy new universal binary to binaryFinalSymlinkLocation
+                copy {
+                    from(tempUniversalBinaryLocation)
+                    into(File(destinationDir, "$frameworkName.framework/Versions/A"))
+                }
+                delete(tempUniversalBinaryLocation)
+                exec {
+                    //recreate the symlink normally to the binary normally in framework folder
+                    commandLine(
+                        "ln",
+                        "-s",
+                        binaryFinalLocation.toPath(),
+                        binaryFinalSymlinkLocation.toPath()
+                    )
+                }
+            }
+        }
+    }
+
+internal fun Project.registerCreateUniversalIosSimulatorFrameworkTask() =
+    task<FatFrameworkTask>("createUniversalIosSimulatorFramework") {
+        group = "multiplatform-swift-package"
+        description = "Creates a universal (fat) ios simulator framework"
+        val configuration = getConfigurationOrThrow()
+        onlyIf { getIosSimulatorFrameworks(configuration).size > 1 }
+        val targets = getIosSimulatorFrameworks(configuration)
+        dependsOn(targets.map { it.linkTask.name })
+        if (targets.isNotEmpty()) {
+            val buildType = if (targets[0].linkTask.name.contains("Release")) "release" else "debug"
+            baseName = configuration.packageName.value
+            destinationDir = buildDir.resolve("bin/iosSimulatorUniversal/${buildType}Framework")
```

Comment:
```
error: the path does not point to a valid framework: xxxx/build/bin/iosSimulatorUniversal/releaseFramework/xxx.framework
```

Reply:
I don't think I touched anything that would affect `outputDirectory`. The intermediate build binaries stored in bin shouldn't affect `outputDirectory`?

How did you cause that error?

Reply:
@luca992 I can provide example gradle file. Version `2.0.3` succeeds. But it seems this fork is generating the intermediate framework with the wrong filename. The output appears to be our manifest package name rather than defined `baseName`.

Reply:
Only seems to be the case for `iosSimulatorUniversal`. Resolution to succeed build is to label `baseName` the same as our manifest package name. But this means the wrong framework name for imports.

Reply:
I hope I understand your problem correctly. Please look at my last commit. I changed name of generated Universal Framework.

Reply:
thanks @jizoquval will take a look this week

Reply:
@markst wait what do you mean `2.0.3` succeeds? Do you mean my `2.0.3-arm64` release? That build does not work at all which is why I made the made a PR for @jizoquval 's PR https://github.com/ge-org/multiplatform-swiftpackage/pull/41#issuecomment-1112860015. `2.0.4-arm64` is the working build

Reply:
haven't been able to test changes yet. gave up trying to use checked out version. might it be possible to make another release on your repository @luca992 ?

Reply:
sure. I'll do that today

Reply:
@markst try my new build `2.0.5-arm64`